### PR TITLE
Added recipe and patch for nesalizer

### PR DESCRIPTION
**Pull request by alwayslivid**
Here's a link to the upstream repository of nesalizer: https://github.com/ulfalizer/nesalizer

Relevant issue: #4439

### CODE REVIEW

File: app-emulation/nesalizer/nesalizer-1.0~git.recipe [hunk 1]
```diff
@@ -0,0 +1,48 @@
+SUMMARY="SDL-based NES emulator"
+DESCRIPTION="A work-in-progress NES emulator with a real-time rewind
+feature that correctly reverses sound."
+HOMEPAGE="https://github.com/ulfalizer/nesalizer"
+COPYRIGHT="2013-2015 Ulf Magnusson"
+LICENSE="GNU GPL v2"
+REVISION="2"
```
Comment (korli) [inline app-emulation/nesalizer/nesalizer-1.0~git.recipe]: REVISION="1"

File: app-emulation/nesalizer/nesalizer-1.0~git.recipe [hunk 2]
```diff
@@ -0,0 +1,48 @@
+SUMMARY="SDL-based NES emulator"
+DESCRIPTION="A work-in-progress NES emulator with a real-time rewind
```
Comment (korli) [inline app-emulation/nesalizer/nesalizer-1.0~git.recipe]: missing backslash at the end of the line

File: app-emulation/nesalizer/nesalizer-1.0~git.recipe (hunk 1 above)
Comment (alwayslivid) [reply app-emulation/nesalizer/nesalizer-1.0~git.recipe]: Got confused by this particular [recipe](https://github.com/haikuports/haikuports/blob/c2a214e6473117adfdbf84d0f792341b30e1bbd7/sys-devel/bc/bc-1.07.1.recipe).

File: app-emulation/nesalizer/nesalizer-1.0~git.recipe [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+BUILD()
+{
+	make CONF=release $jobArgs
+}
+
+INSTALL()
+{
+	make CONF=release INSTALL_DIR=$appsDir
+
+	addAppDeskbarSymlink $appsDir/Nesalizer "Nesalizer"
```
Comment (Begasus) [inline app-emulation/nesalizer/nesalizer-1.0~git.recipe]: I would put this in $prefix/bin ($binDir would resolve to ./self/bin/x86 on gcc2h but we don't want that if gcc2 version can't be build, also drop the symlink, it's a cli binary where you have to pass a <rom> file too.
```diff
-	make CONF=release INSTALL_DIR=$appsDir
-
-	addAppDeskbarSymlink $appsDir/Nesalizer "Nesalizer"
+	make CONF=release INSTALL_DIR=$prefix/bin
```

Comment (alwayslivid) [reply app-emulation/nesalizer/nesalizer-1.0~git.recipe]: The gcc2 version can't be built either way due to compiler incompatibilities, but I'm going to replace this anyways.

Comment (Begasus) [reply app-emulation/nesalizer/nesalizer-1.0~git.recipe]: Hence why we mark it as broken !x86_gcc2, thanks

**Comment by Begasus** [comment]
LGTM @diversys could you give it a check?

**Comment by diversys** [comment]
Do the copyrights really belong to 2013-2015 Ulf Magnusson? I could't find it on Github page.

**Comment by Begasus** [comment]
Seems so @diversys but year should be 2013-2016 (latest commit): https://se.linkedin.com/in/magnussonulf/de

**Comment by alwayslivid** [comment]
> Do the copyrights really belong to 2013-2015 Ulf Magnusson? I could't find it on Github page.

I tried to make use of the commits in order to determine the copyright, but I must have missed something. Will double check just to be sure and correct appropriately.

**Pull request merged**